orttraining/orttraining/python/ort_trainer.py

```diff
@@ -845,6 +870,56 @@ def _prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, i
 
         return input, fetches
 
+    def _debug_conversion_step(self, input, input_descs, output_desc):
```

Comment:
What do you think of something like `_compare_inference_output_between_pytorch_onnx()` to describe what is being done. The public flag already makes it clear that it is a debug flag. This method is internal and can be more accurate in the name